src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -292,20 +293,30 @@ address MacroAssembler::target_addr_for_insn(address insn_addr, unsigned insn) {
   return address(((uint64_t)insn_addr + (offset << 2)));
 }
 
-void MacroAssembler::safepoint_poll(Label& slow_path, bool at_return, bool acquire, bool in_nmethod) {
+void MacroAssembler::safepoint_poll(Label& slow_path, bool at_return, bool acquire, bool in_nmethod, Register scratch) {
```

Comment:
Perhaps rename the `scratch` argument to `tmp` here and below as that's what most of the other MacroAssembler functions are using (and it avoids the visual similarity between `scratch` and `rscratch1`).